modules/spring-config/src/main/java/org/mule/runtime/config/internal/context/lazy/LazyMuleArtifactContext.java

```diff
@@ -319,91 +370,97 @@ public Optional<ComponentModelInitializerAdapter> getParentComponentModelInitial
             .initializeComponents(componentModelPredicate, applyStartPhase));
   }
 
-  private List<Object> createComponents(Optional<Predicate<ComponentAst>> predicateOptional, Optional<Location> locationOptional,
-                                        boolean applyStartPhase,
+  private void validateModel(ArtifactAst artifactAst) throws ConfigurationException {
+    if (dslDeclarationValidationEnabled) {
+      doValidateModel(artifactAst, v -> v.getClass().getAnnotation(IgnoreOnLazyInit.class) == null
+          || v.getClass().getAnnotation(IgnoreOnLazyInit.class).forceDslDeclarationValidation());
+    } else {
+      doValidateModel(artifactAst, v -> v.getClass().getAnnotation(IgnoreOnLazyInit.class) == null);
+    }
+  }
+
+  private void initializeComponentsFromParent(Optional<ComponentModelInitializerAdapter> parentComponentModelInitializerAdapter) {
+    if (parentComponentModelInitializerAdapter.isPresent()) {
+      parentComponentModelInitializerAdapter.get()
+          .initializeComponents(componentModel -> graph.getMissingDependencies()
+              .stream()
+              .anyMatch(missingDep -> missingDep.isSatisfiedBy(componentModel)));
+    } else {
+      graph.getMissingDependencies().stream().forEach(missingDep -> {
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug("Ignoring dependency {} because it does not exist.", missingDep);
+        }
+      });
+    }
+  }
+
+  private void validateRequestedComponentExists(Location location, ArtifactAst minimalArtifactAst) {
+    if (minimalArtifactAst.recursiveStream()
+        .noneMatch(comp -> comp.getLocation() != null
+            && comp.getLocation().getLocation().equals(location.toString()))) {
+      throw new NoSuchComponentModelException(createStaticMessage("No object found at location "
+          + location.toString()));
+    }
+  }
+
+  private void cleanupAndResetComponentsState() {
+    // First unregister any already initialized/started component
+    unregisterBeans(currentComponentInitializationState.getTrackedBeansInOrder());
+
+    objectProviders.clear();
+    resetMuleSecurityManager();
+
+    // This has to be called after all previous state has been cleared because the unregister/cleanup process requires the
+    // errorTypeRepository as it was during its initialization.
+    // TODO: check this, it used to be done with the filtered AST, doing it like this might have a perf impact (not confirmed)
+    // Note that using the full AST directly does not work the same as a FilteredArtifactAst that filters nothing. This
+    // is because the FilteredArtifactAst has some special case for error handlers.
+    doRegisterErrors(new FilteredArtifactAst(getApplicationModel(), c -> true));
```

Comment:
you are using the full ast here, not the minimal one, is that on purpose? (current impl uses the filtered one https://github.com/mulesoft/mule/pull/12183/files#diff-11f3fd72c609272cc90b8d5f05d217084db7d46cdb3421d6ae727412f16c21ceL388)

Reply:
Yes, that is what the TODO is about. I know I changed that. See [this comment](https://github.com/mulesoft/mule/pull/12183#discussion_r1087043103).
If we wanted to use the minimal, we need the minimal that contains both the previous components and the new ones (which we currently don't have). We only have the minimal for the new ones.
I could try to find a solution though if we are worried about registering all the error types.
One quick way out that just came to my mind would be to use the filtered one if doing the old `initializeComponent` methods and use the full one if doing the `initializeAdditionalComponent`.
Ideally I would like to be capable of mutating the error type repository and the error locator so we can register new error types incrementally without having to recreate them fully for each registry miss.

Reply:
As discussed via meet, I changed it now so that the full AST is only used when triggering "additional" components initialization. When doing the regular lazy/selective initialization we do it with the minimal AST as before.